pkgs/servers/home-assistant/custom-components/garmin_connect/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, buildHomeAssistantComponent
+
+, garminconnect
+, tzlocal
+}:
+
+buildHomeAssistantComponent rec {
+  owner = "cyberjunky";
+  domain = "garmin_connect";
+  version = "unstable-2024-01-16";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "home-assistant-garmin_connect";
+    rev = "a12306d79d14d721b8a634820c529825250c736e";
+    sha256 = "sha256-pcJ6y3Ntab7fb0tfvL2HZ78dFQBVdhOhqBhiD3gQac4=";
+  };
+
+  patchPhase = ''
+    # Patch the manifest to allow the newer versions of garminconnect in pkgs
+    sed -i 's/garminconnect==0.2.12/garminconnect>=0.2.15/g' \
+      custom_components/garmin_connect/manifest.json
+
+    # Required for new version of hass
+    # See: https://github.com/home-assistant/example-custom-config/issues/45
+    sed -i \
+      's/\(hass.config_entries.async\)_setup_platforms/await \1_forward_entry_setups/g' \
+      custom_components/garmin_connect/__init__.py
```

Comment:
Please submit that as a patch upstream and fetch it from there.

Reply:
not needed for head